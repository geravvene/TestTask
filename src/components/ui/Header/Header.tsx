import cn from 'classnames/bind';
import style from './header.module.scss';
import useActions from '../../../hooks/useActions';
import { ReactComponent as Light } from '../../../assets/svg/light.svg';
import { ReactComponent as Logo } from '../../../assets/svg/logo.svg';

const cx = cn.bind(style);

interface IHeader {
  isDark: boolean
}

function Header({ isDark }: IHeader) {
  const { change } = useActions();
  return (
    <header
      className={cx('header', {
        dark: isDark,
      })}
    >
      <Logo />
      <button type="button" onClick={() => change()}>
        <Light />
      </button>
    </header>
  );
}
export default Header;
