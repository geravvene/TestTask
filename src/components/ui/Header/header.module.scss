.header {
  width: 100%;
  display: flex;
  padding-top: 20px;
  padding-right: 3rem;
  padding-left: 3rem;
  justify-content: space-between;
  align-items: center;
  background: white;
  svg {
    color: black;
  }
  &.dark {
    background: black;
    svg {
      color: white;
    }
  }
}
