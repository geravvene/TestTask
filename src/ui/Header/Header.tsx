import { useCallback } from 'react';

import cn from 'classnames/bind';

import style from './Header.module.scss';
import useTheme from '@/hooks/useTheme';
import { ReactComponent as Light } from '@svg/light.svg';
import { ReactComponent as Logo } from '@svg/logo.svg';

const cx = cn.bind(style);

function Header() {
  const { isDark, setIsDark } = useTheme();

  const clickChange = useCallback(() => {
    setIsDark(!isDark);
  }, [isDark, setIsDark]);

  return (
    <header
      className={cx('header', {
        dark: isDark,
      })}
    >
      <Logo />
      <button type="button" onClick={clickChange}>
        <Light />
      </button>
    </header>
  );
}
export default Header;
