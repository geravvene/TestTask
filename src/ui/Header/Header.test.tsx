import { render } from '@testing-library/react';
import Header from './Header';
import '@testing-library/jest-dom';

describe('Header component', () => {
  it('renders correctly', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
