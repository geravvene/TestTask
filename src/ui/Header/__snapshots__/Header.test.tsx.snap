// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div>
  <header
    class="header dark"
  >
    <svg />
    <button
      type="button"
    >
      <svg />
    </button>
  </header>
</div>
`;
